content/raw/chat/username_avatar.md

```diff
@@ -0,0 +1,132 @@
+
+### Joining a community
+
+1. User selects a community to join
+2. System detects that another user with the same display name has already joined that community
+3. System warns user that the display name is already taken in that community
+4. System displays conflicting profiles display name, avatar, identicon ring, emoji hash and chat key
+5. User selects "Change Your display name"
+6. User fills in a new display name
+7. Systems adds user to the community
+7.1 alternative: System warns user that the display name is already taken in one of the communities the user is part of
+7.2 alternative: System warns user that the display name is invalid
+
+## Functional Requirements
+
+**Profile Emblems:**
+
+- If a profile is a mutual contact then it MUST display a mutual contact emblem (e.g a profile icon)
+- If a profile is a verified identity then it MUST display a verified emblem (e.g a tick icon)
+- If a profile is marked as untrusthworthy then it MUST display a untrustworthy emblem (e.g a exclamation mark icon)
+- If a profile is a mutual contact then it MAY display another emblem
+- If a profile is not a mutual contact then it MUST NOT display other emblems
+
+**Profile usernames**
+
+- Users MUST be able to set a display name for their own profile
+- Display names MUST be at least 5 characters long
+- Display names MUST be composed of alphanumeric characters, underscores, and/or hyphens
+- Display names MUST NOT be longer than 24 characters
+- Display names MUST NOT be "3 word names"
+- Display names MUST NOT end in "_eth" or ".eth" or "-eth"
+- Profiles containing 3 word names as their display name that does not deterministically correspond to their chatkey MUST display a warning icon to indicate that the username is not deterministic
+- A user MUST be able to set a local username for any profile other than their own
+- A local username MUST be visible only to the user who set it
+- If profile has a local set username set then it MUST show the original display name (e.g if the local username is 'intimidating guy' and the original display name is 'Jotaro' then `intimidating guy (Jotaro)`)
+- If a profile has a registered ENS username, then it MUST end in .eth
+- If a profile has a registered ENS stateofus.eth username, then it MUST end in stateofus.eth
+- If a profile has a registered ENS username, then it MAY NOT display the chat key next to the username
+- If a profile does not have a registered ENS username, then it MUST display the chat key next to the username
+
+**Profile avatar**
+
+- If a profile does not have a profile photo set it MUST use a 2 letter avatar
+- A 2 letter avatar MUST be the first two letters of the display name
+
+- Users MUST be able to set a profile photo for their profile
+- Profile photos MUST be visible to all users
+
+- A identicon ring MUST be composed of 5 color segments
+- A identicon ring segment MAY be one of 32 distinctive colors
+- A identicon ring segment MAY be of different length
```

Comment:
_Suggestion_: It might be more clarifying if we add the word `units` to refer to the length of the segment and as understood from figma, this units length should be from **1 to 5** so, it could be interesting to mention it here, as well. Something like this: `Color Hash segment MAY be of different units length, from 1 to 5 units`

Reply:
We are going to encode 10 bytes of data into identicon ring, which implies: 49 units, 10 segments, 32 colors. A segment can be <1,40> units long (see calculations here: https://github.com/status-im/specs/pull/166)